packages/targeting/src/targeting-idb-store.ts

```diff
@@ -3,110 +3,148 @@ import { DBSchema, IDBPDatabase, IDBPTransaction, openDB } from 'idb';
 export interface TargetingDB extends DBSchema {
   eventTypesForSession: {
     key: number;
     value: {
       sessionId: number;
-      eventTypes: Array<{ event_type: string }>;
+      eventTypes: EventTypeStore;
     };
   };
 }
 
-export const createStore = async (dbName: string) => {
-  return await openDB<TargetingDB>(dbName, 1, {
-    upgrade: (db: IDBPDatabase<TargetingDB>) => {
-      if (!db.objectStoreNames.contains('eventTypesForSession')) {
-        db.createObjectStore('eventTypesForSession', {
-          keyPath: 'sessionId',
-        });
-      }
-    },
-  });
-};
-
-const openOrCreateDB = async (apiKey: string) => {
-  const dbName = `${apiKey.substring(0, 10)}_amp_targeting`;
-  return await createStore(dbName);
-};
-
-export const updateEventListForSession = async ({
-  sessionId,
-  eventType,
-  loggerProvider,
-  tx,
-}: {
-  sessionId: number;
-  eventType: string;
-  loggerProvider: ILogger;
-  tx: IDBPTransaction<TargetingDB, ['eventTypesForSession'], 'readwrite'>;
-}) => {
-  try {
-    const eventTypesForSessionStorage = await tx.store.get(sessionId);
-    const eventTypesForSession = eventTypesForSessionStorage ? eventTypesForSessionStorage.eventTypes : [];
-
-    const updatedEventTypes = eventTypesForSession.concat({ event_type: eventType });
-    await tx.store.put({ sessionId, eventTypes: updatedEventTypes });
-    return updatedEventTypes;
-  } catch (e) {
-    loggerProvider.warn(`Failed to store events for targeting ${sessionId}: ${e as string}`);
-  }
-  return undefined;
-};
-
-export const deleteOldSessionEventTypes = async ({
-  currentSessionId,
-  loggerProvider,
-  tx,
-}: {
-  currentSessionId: number;
-  loggerProvider: ILogger;
-  tx: IDBPTransaction<TargetingDB, ['eventTypesForSession'], 'readwrite'>;
-}) => {
-  try {
-    const allEventTypeObjs = await tx.store.getAll();
-    for (let i = 0; i < allEventTypeObjs.length; i++) {
-      const eventTypeObj = allEventTypeObjs[i];
-      const amountOfTimeSinceSession = Date.now() - eventTypeObj.sessionId;
-      if (eventTypeObj.sessionId !== currentSessionId && amountOfTimeSinceSession > MAX_IDB_STORAGE_LENGTH) {
-        await tx.store.delete(eventTypeObj.sessionId);
-      }
+export class TargetingIDBStore {
+  dbs: { [apiKey: string]: IDBPDatabase<TargetingDB> } | undefined;
+
+  createStore = async (dbName: string) => {
+    return await openDB<TargetingDB>(dbName, 1, {
+      upgrade: (db: IDBPDatabase<TargetingDB>) => {
+        if (!db.objectStoreNames.contains('eventTypesForSession')) {
+          db.createObjectStore('eventTypesForSession', {
+            keyPath: 'sessionId',
+          });
+        }
+      },
+    });
+  };
+
+  openOrCreateDB = async (apiKey: string) => {
+    if (this.dbs && this.dbs[apiKey]) {
+      return this.dbs[apiKey];
+    }
+    const dbName = `${apiKey.substring(0, 10)}_amp_targeting`;
+    const db = await this.createStore(dbName);
+    this.dbs = {
+      ...this.dbs,
+      [apiKey]: db,
+    };
```

Comment:
Nit: Why not?
```suggestion
    this.dbs[apiKey] = db;
```

Reply:
I think I did this in case `this.dbs` wasn't defined, however I'm realizing I can just provide a base empty object for it - will update